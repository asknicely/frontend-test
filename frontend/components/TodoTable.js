import React from 'react';
import TodoListItem from './TodoListItem';
import TodoTableHeader from './TodoTableHeader';

function TodoTable(props) {
  const { title, todoList, handleDeleteTodo } = props;

  return (
    <React.Fragment>
      <h1>{title}</h1>

      <table className="table table-striped">
        <TodoTableHeader />
        <tbody>
          {todoList.length >= 1 && (
            <TodoListItem list={todoList} handleDeleteTodo={handleDeleteTodo} />
          )}
        </tbody>
      </table>
    </React.Fragment>
  );
}

export default TodoTable;
