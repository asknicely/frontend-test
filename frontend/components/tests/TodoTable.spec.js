import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TodoTable from '../TodoTable';
import {
  todos,
  openDeleteConfirmation,
  handlToggleCompleteTask,
} from './helpers/helpers';

const todoList = todos;
const titleListPage = 'List of tasks';

describe('TodoTable', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<TodoTable />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render TodoTable correctly', () => {
    const wrapper = shallow(
      <TodoTable
        title={titleListPage}
        todoList={todoList}
        openDeleteConfirmation={openDeleteConfirmation}
        handlToggleCompleteTask={handlToggleCompleteTask}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
