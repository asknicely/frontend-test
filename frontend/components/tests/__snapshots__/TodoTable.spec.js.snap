// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoTable should render TodoTable correctly 1`] = `ShallowWrapper {}`;

exports[`TodoTable snapshot renders 1`] = `
Array [
  <h1
    className="page-title"
  />,
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          User
        </th>
        <th>
          Description
        </th>
        <th />
      </tr>
    </thead>
  </table>,
]
`;
